src/runtime_src/core/common/api/xrt_hw_context.cpp

```diff
@@ -37,13 +48,103 @@ class hw_context_impl : public std::enable_shared_from_this<hw_context_impl>
   using qos_type = cfg_param_type;
   using access_mode = xrt::hw_context::access_mode;
 
+  // Struct used for initializing and dumping firmware log buffer
+  struct uc_log_buffer
+  {
+    size_t m_num_uc; // number of uc's
+    uint32_t m_slot_idx; // index of slot in which these uc's are present
+    size_t m_size_per_uc;
+    xrt::bo m_uc_log_bo; // log buffer used for uc logging
+
+    static void
```

Comment:
This probably shouldn't be static.  Let this function use the data member, so no need to pass the bo.
Note static is required for auxiliary functions used in constructor initializer list, but this function is not, as far as I can tell.